src/main/java/core/basesyntax/Main.java

```diff
@@ -0,0 +1,23 @@
+package core.basesyntax;
+
+public class Main {
+    public static void main(String[] args) {
+        ColorSupplier colorSupplier = new ColorSupplier();
+        FigureSupplier figureSupplier = new FigureSupplier(colorSupplier);
+
+        Figure[] figures = new Figure[6];
+
+        for (int i = 0; i < figures.length; i++) {
+            Figure figure;
+            if (i < 3) {
+                figure = (Figure) figureSupplier.getRandomFigure();
+                figure.draw();
+            } else {
+                figure = (Figure) figureSupplier.getDefaultFigure();
+                figure.draw();
+            }
+            figures[i] = (Figure) figure;
+        }
```

Comment:
```suggestion
        for (int i = 0; i < 6; i++) {
            Figure figure;
            if (i < 3) {
                figure = (Figure) figureSupplier.getRandomFigure();
            } else {
                figure = (Figure) figureSupplier.getDefaultFigure();
            }
            figure.draw();
        }
```

Reply:
![image](https://github.com/mate-academy/jv-oop-advanced/assets/125391179/06945fc0-c0f2-45ac-8a58-b2a046325269)

It looks fixed on my side...